File: C/Searching/Searching_01.c
/*
===============================================================================
Program Name    : Searching Skeleton
Program Number  : 001
File Name       : Searching01.c

Description     : Basic Skeleton Program for Searching Techniques.

Date            : August 2026    
Language        : C
Author          : Atharv Tushar Bhosale
===============================================================================
*/

#include<stdio.h>
#include<stdlib.h>

struct Searching
{
    int *Arr;
    int iSize;
};

typedef struct Searching SEARCHING;
typedef struct Searching* PSEARCHING;

void Accept(PSEARCHING ptr)
{
}

void Display(PSEARCHING ptr)
{
}

int main()
{
    SEARCHING sobj;

    sobj.iSize = 5;
    sobj.Arr = (int *)malloc(sizeof(int) * sobj.iSize);

    Accept(&sobj);
    Display(&sobj);

    free(sobj.Arr);

    return 0;
}

/*
===============================================================================
Output :

(No Output)

===============================================================================
End of Program
===============================================================================
*/

File: Java/Searching/Searching_01.java
/*
===============================================================================
Program Name    : Searching Skeleton
Program Number  : 001
File Name       : Searching_01.java

Description     : This program demonstrates the basic
                  structure of Searching Application.

Concepts Used:
1. Class and Object
2. Dynamic Memory Allocation
3. Array Handling

Date            : August 2026
Language        : Java
Author          : Atharv Tushar Bhosale
===============================================================================
*/

class Searching
{
    private int Arr[];
    private int iSize;

    public Searching(int iNo)
    {
        iSize = iNo;
        Arr = new int[iSize];
    }
}

class Searching_01
{
    public static void main(String A[])
    {
        Searching sobj = new Searching(5);

        System.out.println("Searching object created successfully");
    }
}

/*
===============================================================================
Sample Output

Searching object created successfully

===============================================================================
End of Program
===============================================================================
*/
